library/src/scripts/theming/ThemesActions.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license Proprietary
+ */
+
+import ReduxActions, { bindThunkAction, useReduxActions } from "@library/redux/ReduxActions";
+import { actionCreatorFactory } from "typescript-fsa";
+import { IApiError } from "@library/@types/api/core";
+import { ITheme } from "@library/theming/themeReducer";
+
+const actionCreator = actionCreatorFactory("@@themes");
+
+export enum ThemeType {
+    DB = "themeDB",
+    FS = "themeFile",
+}
+
+export enum PreviewStatusType {
+    PREVIEW = "preview",
+    APPLY = "apply",
+    CANCEL = "cancel",
+}
+export interface IManageTheme extends ITheme {
+    name: string;
+    type: ThemeType;
+    current: boolean;
+    themeID: string;
+    parentTheme: string;
+    parentVersion: string;
+    preview: {
+        [key: string]: string;
+    };
+}
+
+type IGetAllThemeResponse = IManageTheme[];
+
+interface IPutCurrentThemeRequest {
+    themeID: number | string;
+    type: PreviewStatusType;
+}
+/**
+ * Actions for working with resources from the /api/v2/knowledge-bases endpoint.
+ */
+
+export default class ThemesActions extends ReduxActions {
+    public static readonly getAllThemes_ACS = actionCreator.async<{}, IGetAllThemeResponse, IApiError>(
+        "GET_ALL_THEMES",
+    );
+
+    public static readonly putCurrentThemeACs = actionCreator.async<
+        { themeID: number | string },
+        IManageTheme,
+        IApiError
+    >("PUT_CURRENT");
+
+    public static readonly putPreviewThemeACs = actionCreator.async<IPutCurrentThemeRequest, IManageTheme, IApiError>(
+        "PUT_PREVIEW",
```

Comment:
We have a `ThemeActions` right next to this one. It seems really confusing to have both. Could we merge this into that one?